app/services/builder_base_service.rb

```diff
@@ -0,0 +1,25 @@
+class BuilderBaseService
+  def self.build
+    new
+  end
+
+  def self.call(*args, **kwargs)
+    inst = build
+    result = nil
+    errors = []
+
+    # Establish basic interface
+    unless inst.respond_to?(:call)
+      raise NotImplementedError.new("Contract violation, Builder Service Children must implement call")
+    end
+
+    # Error handling by default
+    begin
+      result = inst.call(*args, **kwargs)
+    rescue => e
+      errors.append(e)
+    end
+
+    OpenStruct.new(success?: errors.empty?, errors: errors, result: result)
+  end
```

Comment:
@tsmartt My core problem right now with the the instance based `.call` method (i.e. `Service.build.call` ) is that we lose some of the coherence in the `Class.call` api.

When using the `call` method alone we gaurantee that all children must implement an instance call, can optionally override the class `build`, and all calls to `Class.call` will never raise an exception and will always return the same response.  All I have to do as a developer is deal with my childclass and return a result of any type and it will be reported. 

If we use `Service.build.call` we loose this because we are now faced with a few possibilities:
1. We override `Service.new.call` which means we loose the potency of the interface and the best we can do is abstract out the error handling and OpenStruct methods into some helper methods, but it then is left to the developer to know about/use those methods.
2. We define a BaseClass `BaseService.new.call` method that implements error handling and return values and then we use some new method in the child class to implement the primary API. `run` or some such.

I truly don't care about whether or not we call `build` in production, but I do care that the primary `call` method defines a coherent interface that has to be explicitly overridden in any given subclass rather than explicitly defined in all subclasses.